code/game/machinery/telecomms/presets.dm

```diff
@@ -175,6 +175,88 @@
 	freq_listening = list(COLONY_FREQ)
 	faction_shorthand = "colony"
 
+GLOBAL_LIST_EMPTY(all_static_telecomms_towers)
+
+/obj/structure/machinery/telecomms/relay/preset/tower/Initialize()
+	GLOB.all_static_telecomms_towers += src
+	. = ..()
+
+/obj/structure/machinery/telecomms/relay/preset/tower/destroy()
```

Comment:
Mind that `/datum/proc/Destroy` and `/obj/structure/proc/destroy` are two different procs -- and i think what you want is the former